core/app/[locale]/(default)/account/addresses/_actions/update-address.ts

```diff
@@ -0,0 +1,272 @@
+import { BigCommerceAPIError } from '@bigcommerce/catalyst-client';
+import { parseWithZod } from '@conform-to/zod';
+import { unstable_expireTag as expireTag } from 'next/cache';
+import { getTranslations } from 'next-intl/server';
+import { z } from 'zod';
+
+import { Field, FieldGroup } from '@/vibes/soul/primitives/dynamic-form/schema';
+import { schema } from '@/vibes/soul/sections/address-list-section/schema';
+import { getSessionCustomerAccessToken } from '~/auth';
+import { client } from '~/client';
+import { graphql, VariablesOf } from '~/client/graphql';
+import { TAGS } from '~/client/tags';
+import { FieldNameToFieldId } from '~/data-transformers/form-field-transformer/utils';
+
+import { type State } from './address-action';
+
+export const UpdateCustomerAddressMutation = graphql(`
+  mutation UpdateCustomerAddressMutation($input: UpdateCustomerAddressInput!) {
+    customer {
+      updateCustomerAddress(input: $input) {
+        errors {
+          __typename
+          ... on AddressDoesNotExistError {
+            message
+          }
+          ... on CustomerAddressUpdateError {
+            message
+          }
+          ... on CustomerNotLoggedInError {
+            message
+          }
+          ... on ValidationError {
+            message
+            path
+          }
+        }
+        address {
+          entityId
+        }
+      }
+    }
+  }
+`);
+
+const stringToNumber = z.coerce.string().pipe(z.coerce.number());
+
+const inputSchema = z.object({
+  addressEntityId: stringToNumber,
+  data: z.object({
```

Comment:
The only big difference between this schema and the add address schema is that the fields are optional and are nested in this `data` key.